library/L1_Peripheral/lpc40xx/capture.hpp

```diff
@@ -0,0 +1,285 @@
+            .isr = TimerHandler<kTimerPartial2>
+        };
+        inline static const CaptureChannel_t kCaptureTimer3 =
+        {
+            .channel = kTimerPartial3,
+            .isr = TimerHandler<kTimerPartial3>
+        };
+    };  // struct Channel
+
+
+    static void TimerHandler(const CaptureChannelPartial_t & channel)
+    {
+        constexpr bit::Mask kCaptureInterruptFlagBit =
+            bit::CreateMaskFromRange(4, 5);
+        constexpr uint8_t kResetCaptureInterrupts = 0b11;
+
+        if (*channel.user_callback != nullptr)
+        {
+            (*channel.user_callback)();
+        }
+
+        channel.timer_register->IR = bit::Insert(channel.timer_register->IR,
+            kResetCaptureInterrupts, kCaptureInterruptFlagBit);
+    }
+
+    struct CaptureStatus_t
+    {
+        uint32_t count;
+        uint32_t flags;
+    };
+
+    static constexpr sjsu::cortex::InterruptController kInterruptController =
+        sjsu::cortex::InterruptController();
+
+    explicit constexpr Capture(
+        const CaptureChannel_t &timer,
+        const sjsu::SystemController &system_controller =
+            DefaultSystemController(),
+        const sjsu::InterruptController &interrupt_controller =
+            kInterruptController
+        )
+        : timer_(timer),
+          system_controller_(system_controller),
+          interrupt_controller_(interrupt_controller)
+    {
+    };
+
+    Status Initialize(
+        units::frequency::hertz_t frequency,
+        IsrPointer isr = nullptr, int32_t priority = -1) const override
+    {
+        // NOTE: Same initialization as Timer library's Initialize()
+        system_controller_.PowerUpPeripheral(timer_.channel.power_id);
+        SJ2_ASSERT_FATAL(
+            frequency != 0_Hz,
+            "Cannot have zero ticks per microsecond, please choose 1 or more.");
+
+        // Set Prescale register for Prescale Counter to milliseconds
+        uint32_t prescaler =
+            system_controller_
+                .GetPeripheralFrequency(timer_.channel.power_id) / frequency;
+        timer_.channel.timer_register->PR = prescaler;
+
+        // Enable timer
+        constexpr bit::Mask kTimerEnableBit = bit::CreateMaskFromRange(0);
+        timer_.channel.timer_register->TCR = bit::Set(
+                timer_.channel.timer_register->TCR, kTimerEnableBit.position);
+
+        // Install capture ISR
+        *timer_.channel.user_callback = isr;
+        interrupt_controller_.Register({
+                .interrupt_request_number = timer_.channel.irq,
+                .interrupt_service_routine = timer_.isr,
+                .enable_interrupt = true,
+                .priority = priority
+        });
+
+        // Disable both capture channels
+        ConfigureCapture(kCaptureChannel0, kCaptureEdgeModeNone);
+        ConfigureCapture(kCaptureChannel1, kCaptureEdgeModeNone);
+        EnableCaptureInterrupt(kCaptureChannel0, false);
+        EnableCaptureInterrupt(kCaptureChannel1, false);
```

Comment:
Should you do this first before you register the ISR. If these are defined as true before you get to this point, there is a chance that an event can occur between the registration of the ISR and this disabling of the ISR. This is more evident in a threaded system where the task controlling this peripheral is of low priority and not getting as much CPU time.